pkg-manager/client/src/index.ts

```diff
@@ -56,13 +57,13 @@ type Fetchers = {
 function createFetchers (
   fetchFromRegistry: FetchFromRegistry,
   getAuthHeader: GetAuthHeader,
-  opts: Pick<ClientOptions, 'rawConfig' | 'retry' | 'gitShallowHosts' | 'resolveSymlinksInInjectedDirs' | 'unsafePerm'>,
+  opts: Pick<ClientOptions, 'rawConfig' | 'retry' | 'gitShallowHosts' | 'resolveSymlinksInInjectedDirs' | 'unsafePerm' | 'includeOnlyPackageFiles'>,
   customFetchers?: CustomFetchers
 ): Fetchers {
   const defaultFetchers = {
     ...createTarballFetcher(fetchFromRegistry, getAuthHeader, opts),
     ...createGitFetcher(opts),
-    ...createDirectoryFetcher({ resolveSymlinks: opts.resolveSymlinksInInjectedDirs }),
+    ...createDirectoryFetcher({ resolveSymlinks: opts.resolveSymlinksInInjectedDirs, includeOnlyPackageFiles: opts.includeOnlyPackageFiles }),
```

Comment:
Did you meaning this case ？
```
 // @pkg/bar
 {
    name: "@pkg/bar"
 }
 // @pkg/foo
 {
   name: "@pkg/foo",
   "dependencies": {
     "@pkg/bar": "workspace:*",
     "baz": "file:../../baz"
   }
 }
```
if we set `opts.includeOnlyPackageFiles: true`, it will affect file protocol dependency(`baz`) too.

If only local dependencies (workspace protocol, file protocol, link protocol) use the directory fetcher, it seems to be ok to follow the files setting in package.json by default (if someone need to copy all files, then remove the files setting).
I'm not sure if there are other scenarios that are affected that I'm missing, if so, please let me know, thanks a lot.